src/main/java/com/uid2/admin/audit/QLDBInit.java

```diff
@@ -1,350 +1,134 @@
+            qldbDriver.execute(txn -> {
+                Result result = txn.execute("SELECT COUNT(*) AS \"count\" FROM " + qldbTableName
+                        + " AS t WHERE t.data.itemType = ?", ionSys.newString(service.tableType().toString()));
+                count[0] = (IonStruct) result.iterator().next();
+            });
+            return !count[0].get("count").equals(ionSys.newInt(0));
         } catch (Exception e) {
-            e.printStackTrace();
+            throw new RuntimeException("AWS configuration not set up");
         }
     }
 
-    private static void addEnclaveModels(Collection<Collection<OperationModel>> models, EnclaveIdentifierProvider enclaveIdProvider) {
+    private static void createTable() { //creates the logs table. Assumes it doesn't already exist
         try {
-            Collection<EnclaveIdentifier> enclaves = enclaveIdProvider.getAll();
-            Collection<OperationModel> newModels = new HashSet<>();
-            for (EnclaveIdentifier e : enclaves) {
-                newModels.add(new OperationModel(Type.ENCLAVE, e.getName(), null,
-                        DigestUtils.sha256Hex(jsonWriter.writeValueAsString(e)), null));
-            }
-            models.add(newModels);
+            qldbDriver.execute(txn -> {
+                txn.execute("CREATE TABLE " + qldbTableName);
+            });
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
 
-    private static void addSaltModels(Collection<Collection<OperationModel>> models, RotatingSaltProvider saltProvider) {
+    private static void createIndices() { //creates indices on itemType and itemKey. Assumes they don't already exist.
         try {
-            List<RotatingSaltProvider.SaltSnapshot> snapshots = saltProvider.getSnapshots();
-            RotatingSaltProvider.SaltSnapshot lastSnapshot = snapshots.get(snapshots.size() - 1);
-
-            JsonObject jo = new JsonObject();
-            jo.put("effective", lastSnapshot.getEffective().toEpochMilli());
-            jo.put("expires", lastSnapshot.getExpires().toEpochMilli());
-            jo.put("salts_count", lastSnapshot.getAllRotatingSalts().length);
-            jo.put("min_last_updated", Arrays.stream(lastSnapshot.getAllRotatingSalts())
-                    .map(SaltEntry::getLastUpdated)
-                    .min(Long::compare).orElse(null));
-            jo.put("max_last_updated", Arrays.stream(lastSnapshot.getAllRotatingSalts())
-                    .map(SaltEntry::getLastUpdated)
-                    .max(Long::compare).orElse(null));
-
-            Collection<OperationModel> newModels = new HashSet<>();
-            newModels.add(new OperationModel(Type.SALT, "singleton", null,
-                    DigestUtils.sha256Hex(jo.toString()), null));
-            models.add(newModels);
+            qldbDriver.execute(txn -> {
+                txn.execute("CREATE INDEX ON " + qldbTableName + "(itemType)");
```

Comment:
why don't create one index on two fields

Reply:
QLDB does not support an index on multiple fields